contracts/registry/src/contract.rs

```diff
@@ -222,21 +239,30 @@ fn query_contracts(deps: Deps) -> StdResult<ContractsResponse> {
                 is_executable: contract.is_executable,
             })
         })
+        .take(limit)
```

Comment:
Shouldn't the `.take(limit)` be invoked before the `.map`? @maurolacy 

Otherwise, wouldn't the map closure fn be applied to all items in the range (and we just take the first `limit` results of that?

Reply:
Anyways, I've refactored it now